h2o-r/h2o-package/R/explain.R

```diff
@@ -230,6 +230,10 @@ case_insensitive_match_arg <- function(arg, choices) {
         }
         x <<- single_model@allparameters$x
         y <<- single_model@allparameters$y
+        treatment <- single_model@allparameters$treatment_column
```

Comment:
This would work only for single models or list of models where the first model is uplift...

I would suggest to do something like
```diff
diff --git a/h2o-r/h2o-package/R/explain.R b/h2o-r/h2o-package/R/explain.R
index d228c5c4b8..a7c729c39f 100644
--- a/h2o-r/h2o-package/R/explain.R
+++ b/h2o-r/h2o-package/R/explain.R
@@ -242,7 +242,10 @@ case_insensitive_match_arg <- function(arg, choices) {
       .self
     },
     get_model = function(model_id) {
-      return(memoised_models$get_model(model_id))
+      model <- memoised_models$get_model(model_id)
+      if (is.null(model@allparameters$treatment_column))
+        stop("Uplift is not supported in explain yet.")
+      return(model)
     }
   )
 )

```

R h2o models are pretty expensive when it comes to creation as we download all the info and fill the objects (unlike python where you have can have properties that can download the info on demand) so we cache the models that were used so we don't materialize them multiple times (explain creates a varimp plot, pdp, shap etc and with caching we materialize the model only once). So IIRC the `get_model` method should be used everywhere in explain.